Hmwk_#3_Hangman/hangman_game.py

```diff
@@ -0,0 +1,159 @@
+def draw_mask(word_to_guess, letter_approval, guesses, placeholder, letter):
+   
+    word_placeholder = placeholder
+    letters_to_guess = [letter for letter in word_to_guess]
+    if letter_approval == 'Y':
+        letter_indices = [ind for ind, character in enumerate(letters_to_guess) if character == letter] #receive index where guessed letter shoul be stored
+        for i in letter_indices:
+           
+            word_placeholder[i] = ' ' + letter + ' ' #set guessed letter on it's place in the word
+    mask = '\n' + ' '.join(word_placeholder) + '\n'
+    print(mask)
+
+    return word_placeholder
+
+#run the flow
+def play_game(filename):
+
+    logo = hangman_graphics.get_logo() #add some graphics
+    print(logo)
+    game_mode = set_game_mode() #define mode
+    image = hangman_graphics.get_image() #add graphics for each turn
+    words = get_words(filename) #receive string from file
+    characters = words #use string copy for letters frequency check without reopening the file
+    word_to_guess = choose_word(words) #define word
+    letter_freq = get_letter_freq(characters) #get dictionary of characters and their frequency
+
+    guesses = 0 
+    fail_counter = 0 #add counter to draw appropriate 'hangman' graphics if letter was guessed after a fail turn
+    word_placeholder = initial_mask(word_to_guess)
+
+    while guesses <= 10: #player have 10 tries
+        print("Word to guess: ", word_to_guess)
+        print("Guess try #", guesses, '\n')
+        letter = pick_up_letter(letter_freq) 
+        letter_approval = letter_appearance(game_mode, word_to_guess, letter) #check if letter exists in word
+        letter_freq.pop(letter) # remove letters from end of dictionary (ascending sort allows to move form most frequent to lower ones)
+                                # despite if it was guessed or not, letter cannot be used twice
+        guesses = guess_counter(guesses)
+        time.sleep(1.5) #add delay for realistic rendering
+        draw_mask(word_to_guess, letter_approval, guesses, word_placeholder, letter) #render word placeholder with guessed and yet not guessed letters
+
+        if letter_approval == "Y":
+            for i in image[fail_counter]: #render previous failed 'hangman' step, failed turns shoul not rollback 
+                print(i)
+        elif letter_approval == "N":
+            fail_counter += 1
+            for i in  image[fail_counter]: #render next failed 'hangman' step, + 1 step to loss
+                print(i)
+        if guesses == 10 and word_placeholder != [ch for ch in word_to_guess]:
+            print("You couldn't guess the word!")
+        elif word_placeholder == [ch for ch in word_to_guess]:
+            print("Hooray, congrats, you've guessed the word!") 
+        guesses += 1
+        print("=================================================")
+        time.sleep(5) #add delay for realistic rendering of turns
+
+
+
+play_game("words.txt")
```

Comment:
PEP8 recommends leaving a blank line at the end of a .py file

https://pep8.readthedocs.io/en/release-1.7.x/intro.html